csrc/kernel_cache.cpp

```diff
@@ -350,8 +359,64 @@ FusionKernelRuntime* FusionExecutorCache::getKernelRuntimeFor(
     }
   }
 
-  // Access kernels associated with the common device id
-  auto& kernel_runtimes = kernel_runtimes_[args.getDeviceIndex()];
+  auto getConcretizationInfoFromArgs = [args](Fusion* fusion) {
+    ExpressionEvaluator expr_eval;
+
+    // Bind input scalars and tensor metadata to symbolic scalars
+    // Here we bind only the inputs that are needed to concretize dynamic
+    // transforms.
+    TORCH_CHECK(
+        args.size() == fusion->inputs().size(),
+        "Received ",
+        args.size(),
+        " inputs but expected ",
+        fusion->inputs().size());
+    for (auto i : c10::irange(args.size())) {
+      auto inpi = fusion->inputs()[i];
+      if (inpi->isIntegralScalar()) {
+        TORCH_CHECK(
+            args[i]->isType(ArgType::Long),
+            "Expected integer input at position ",
+            i,
+            " but found ",
+            argTypeToString(args[i]->type()));
+
+        const int64_t arg_val =
+            *reinterpret_cast<const int64_t*>(args[i]->arg());
+        expr_eval.bind(inpi, arg_val);
+      } else if (inpi->isA<TensorView>()) {
+        auto tv = inpi->as<TensorView>();
+        auto root = tv->domain()->getRootDomain();
```

Comment:
Also, while it doesn't matter, it's generally more robust to use `getMaybeRfactorDomain` as the actual shape of a tensor. Since this is an input to the original fusion, no tensor should have an rfactor domain, but that may change in the future.